src/main/java/io/github/apickledwalrus/skriptgui/gui/events/GUIEvents.java

```diff
@@ -151,4 +156,36 @@ public void onInventoryClose(InventoryCloseEvent e) {
 		}
 	}
 
+	private void handleDoubleClick(GUI gui, InventoryClickEvent event) {
+		GUIEventHandler eventHandler = gui.getEventHandler();
+
+		Inventory guiInventory = gui.getInventory();
+		int size = guiInventory.getSize();
+		ItemStack cursor = event.getCursor();
+
+		if (cursor == null || event.getCurrentItem() != null)
+			return;
+
+		int totalAmount = cursor.getAmount();
+		List<InventoryClickEvent> clickEvents = new ArrayList<>();
+		for (int slot = 0; slot < size; slot++) {
+			ItemStack item = guiInventory.getItem(slot);
+			if (item != null && item.isSimilar(cursor)) {
+				if (!gui.isRemovable(gui.convert(slot))) {
+					event.setCancelled(true);
+					clickEvents.clear();
+					break;

```

Comment:
```suggestion
					return;

```
This is probably simpler